web/app.js

```diff
@@ -987,10 +993,7 @@ var PDFViewerApplication = {
       pdfViewer.setPageLabels(labels);
       pdfThumbnailViewer.setPageLabels(labels);
 
-      self.hasPageLabels = true;
-      self._updateUIToolbar({
-        resetNumPages: true,
-      });
+      self.toolbar.setPagesCount(pdfDocument.numPages, true);
```

Comment:
You need to call the `{toolbar, secondaryToolbar}.setPageNumber(...)` methods here too, otherwise the page number input will be empty when loading e.g. http://mirrors.ctan.org/info/lshort/english/lshort.pdf.